### PR TITLE
Port ImageResource to params and results

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

This starts the work on using `params` for specifying the image reference, and result (when supported) to store the digest.

The `kaniko` is an example with the `results`.

Depends on tektoncd/pipeline#2212

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Yaml file complies with [yamllint](https://github.com/adrienverge/yamllint) rules.

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: kaniko/kaniko.yaml [hunk 1]
```diff
@@ -35,7 +36,17 @@ spec:
...
       runAsUser: 0
+  - name: write-digest
+    workingDir: $(workspaces.sources.path)
+    image: imagedigestexport-and-jq # FIXME(vdemeester) change with updated image
+    script: |
+      # output of imagedigestexport [{"name":"image","digest":"sha256:eed29..660"}]
+      /ko-app/imagedigestexporter \
+        -images [{"name":"$(params.IMAGE)","type":"image","url":"$(params.IMAGE)","digest":"","OutputImageDir":"$(workspaces.source.path)/image-digest"}] \
+        -terminationMessagePath=/tmp/digest
+      cat  | jq '.[0].digest' | tee /tekton/results/IMAGE_DIGEST
```
Comment (sbwsg) [inline kaniko/kaniko.yaml]: Does `jq` add any newlines after the digest string when it's piped like this?  Suggest adding a `tr -d '\n'` if so to make sure the result the controller receives is just the digest value minus whitespace.

Comment (sbwsg) [reply kaniko/kaniko.yaml]: also, should this be `cat /tmp/digest`?

File: kaniko/kaniko.yaml [hunk 2]
```diff
@@ -35,7 +36,17 @@ spec:
...
       runAsUser: 0
+  - name: write-digest
+    workingDir: $(workspaces.sources.path)
+    image: imagedigestexport-and-jq # FIXME(vdemeester) change with updated image
+    script: |
+      # output of imagedigestexport [{"name":"image","digest":"sha256:eed29..660"}]
+      /ko-app/imagedigestexporter \
+        -images [{"name":"$(params.IMAGE)","type":"image","url":"$(params.IMAGE)","digest":"","OutputImageDir":"$(workspaces.source.path)/image-digest"}] \
+        -terminationMessagePath=/tmp/digest
+      cat /tmp/digest | jq '.[0].digest' | tee /tekton/results/IMAGE_DIGEST
```
Comment (afrittoli) [inline kaniko/kaniko.yaml]: The write-digest step could write the output to a file and the next task that runs alpine:python could publish the digest from the JSON to results.

Comment (vdemeester) [reply kaniko/kaniko.yaml]: A task or a step, indeed :wink:

Comment (afrittoli) [reply kaniko/kaniko.yaml]: Step, indeed, that's what I meant

File: kaniko/kaniko.yaml (hunk 1 above)
Comment (afrittoli) [reply kaniko/kaniko.yaml]: > Does `jq` add any newlines after the digest string when it's piped like this? Suggest adding a `tr -d '\n'` if so to make sure the result the controller receives is just the digest value minus whitespace.

`jq -c ` should do the trick too

**Comment by vdemeester** [comment]
It is green :tada:

**Pull request merged**